feat: Реализована работа с кастомными перемеными

LilKirill00 opened the pull request:
можно запросить текст у пользователя через save_said
можно использовать полученые данные в exec_button

LilKirill00 committed:
feat: Реализована работа с кастомными перемеными

можно запросить текст у пользователя через save_said
можно использовать полученые данные в exec_button

polyn0m commented on README.md:
```diff
@@ -403,6 +406,33 @@ echo -n $2 $3
+      - button:
+          id: 100
+          text: "Узнать погоду"
+          save_said: 
```
Я бы лучше этот блок назвал **save_to_var**, а внутри **save_to_var** -> **var_name**

LilKirill00 committed:
Множественые изменения в коде

style: переименовано SaveToVar -> VarName
style: переименовано SaveAdd -> SaveToVar
refactor: перенесен type Chat models.go -> bot.go изза циклического импорта при реализации нового модификатора
feat: реализован модификатор DoButton для SaveToVar
feat: добавление больше функций для работы с данными в cache
refactor: добавление рекурсивной проверки всех кнопок и уровней меню
refactor: добавление подсказки в ошибку на каком уровне находится неправильная часть конфигурации
refactor: перенесено заполнение шаблонов в отдельную функцию
feat: добавлено больше мест где можно использовать шаблоны
docs: улучшение Readme

LilKirill00 committed:
refactor: убран модификатор и улучшена обработка

убран модификатор goto, тк он отлично работает в do_button

добавлены обработки ошибок для save_to_var
исправлены неработающие параметры и модификаторы в do_button

LilKirill00 committed:
refactor: переработка обработки nested menu

разрешено в do_button использовать menu
теперь все nested menu проходят обработку

LilKirill00 committed:
refactor: перенос в другую функцию

перенос части кода в другую функцию чтобы 1 функция отвечала только за настройки а другая только за проверку
